lexer/Cargo.toml

```diff
@@ -0,0 +1,10 @@
+[package]
+name = "lexer"
+version = "0.1.0"
+edition = "2024"
```

Comment:
Can this be set at the workspace level ?

Reply:
It looks like it: https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

Reply:
Ok, I will fix this tomorrow. Can I take the opportunity to move the files so I can remove the `lexer::` prefix? That feels silly. The crate should be just called `lex`